backend/remote/remote.go

```diff
@@ -162,6 +164,7 @@ func (rf *Factory) Create(address string) (types.Backend, error) {
 	}
 
 	rpc := rpc.NewClient(conn)
+	r.replicaDataAddr = conn.RemoteAddr().String()
```

Comment:
In fact I was thinking in https://github.com/rancher/longhorn/blob/master/rpc/client.go#L23 , add a field for peer addr, instead of in replica.

As long as we have conn, we should able to figure peer's address. Also the same field can be by e.g. https://github.com/rancher/longhorn/blob/master/rpc/client.go#L99 .